### ISSUE TITLE
v2.1.0: mpi_open_port does not give any tcp info in portname

### ISSUE THREAD

**Issue by tjb900** [open-mpi/ompi#3322]
Hi!

I'm starting to develop a master/slave app, where a number of slaves will connect to a long-running parallel master process to obtain work and submit the results.  The slaves will be started independently of the master, in separate jobs via a queueing system, and then attempt to connect back to the master using the port name it establishes before submitting the slave jobs.

In a few toy tests of the required componentry, after upgrading to v2.x I seem to be finding that my simple mpi_comm_accept/mpi_comm_connect example no longer generates port names with enough information for processes to connect from outside a job.

In v1.6.5 (with ip addresses/usernames/hostnames replaced):

```
[user@host masterslave_experiments]$ mpirun -np 1 ./test3
Waiting for connection on portname '140771328.0;tcp://X.X.X.X:38508;tcp://X.X.X.X:38508;tcp://X.X.X.X:38508+140771329.0;tcp://X.X.X.X:35340;tcp://X.X.X.X:35340;tcp://X.X.X.X:35340:300'
... (I run another, separate, mpirun which connects to that port name - which I manually copy and paste for this test)
... (then they both succeed and communicate)
```

In v2.0.2rc4 (compiled with default options) and v2.1.0 (built with mpi_thread_multiple), this happens:

```
[user@host masterslave_experiments]$ mpirun -np 1 ./test3
Waiting for connection on portname '1751646209.0:2576713416'
... (I run the other, separate, mpirun with that port name, and they both time out)
```

This occurs both from a remote host (unsurprisingly - I can't see how there's enough info in that port name for one process to find the other) and even on the same host.

Am I missing an MCA parameter or something else required to make this work?

Thanks!
Tim

**Comment by tjb900** [comment]
My apologies, it appears that this could be a duplicate of #2363.  That said, that issue is closed but not fixed!

**Comment by tjb900** [comment]
I think it was mentioned in #2363 that a fix was imminent for this, then expected in v2.0.2.  However, I can replicate this on master as well as v2.0.x and can't find any PRs that seem (to my inexperienced eye) to relate.  Any insight would be greatly appreciated!

Thanks very much,
Tim

**Comment by rhc54** [comment]
It is still on the list - just low in priority. It should be fixed for v3.0, but I honestly don't know if I'll find time to backport it.

**Comment by tjb900** [comment]
@rhc54 Thanks for the update.

**Comment by rhc54** [comment]
I looked into this and provided a fix for master and v3.x. Bottom line is that you won't get a TCP-like port name as we don't use TCP to support connect/accept operations. Instead, you'll need to ensure that ompi-server is running and that both jobs have access to it. I've added some detection in case someone passes in (e.g., off a command line) an incorrect port string, or ompi-server isn't available, so we can print out an error message explaining what needs to be done to make it work.

I'm afraid there are no plans to backport this to the 2.x series.
